inc/saip4ext.h

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 20XX Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saip4ext.h
+ *
+ * @brief   This module defines SAI P4 Extension interface
+ */
+
+#if !defined (__SAIP4EXT_H_)
+#define __SAIP4EXT_H_
+
+#include <saitypes.h>
+
+/**
+ * @brief Attribute Id for P4 ext
+ */
+typedef enum _sai_p4ext_entry_attr_t
+{
+    /**
```

Comment:
We might need saiswitch attribute to check if SAI adapter suppots P4 Extension or not.